pkg/envoy/cds/cluster.go

```diff
@@ -25,14 +25,30 @@ const (
 	clusterConnectTimeout = 1 * time.Second
 )
 
+type options struct {
+	noPermissive bool
+	noTLS        bool
```

Comment:
Let's invert these to `permissive, withTLS`

Reply:
that changes the default values to false, is that something we want?

Reply:
There's no default in the current impl, just based on what the configurator interface returns. Should be fine to change these to `permissiveMode, withTLS`  and remove the usage of configurator within this function for permissive mode.

Reply:
@steeling I am assuming the plan is to still invert negatives.

Reply:
i meant the default for the call to the unimported function, but happy to make that change!